catalyst-toolbox/src/stats/voters.rs

```diff
@@ -36,21 +33,86 @@ pub fn calculate_wallet_distribution<S: Into<String>>(
                 chain_addr::Address(discrimination, Kind::Account(public)),
             )
         })
-        .collect();
+        .collect()
+}
+
+fn vote_counts_as_addresses(
+    votes_count: VoteCount,
+    genesis: &Block0Configuration,
+) -> Vec<(InitialUTxO, u32)> {
+    votes_count
+        .iter()
+        .filter_map(|(address, votes_count)| {
+            for initials in &genesis.initial {
+                if let Initial::Fund(funds) = initials {
+                    if let Some(utxo) = funds.iter().find(|utxo| {
+                        account_hex_to_address(
+                            address.to_string(),
+                            genesis.blockchain_configuration.discrimination,
+                        )
+                        .unwrap()
+                            == utxo.address
+                    }) {
+                        return Some((utxo.clone(), *votes_count as u32));
+                    }
+                }
+            }
+            None
+        })
+        .collect()
```

Comment:
Reversing this iteration would be much faster. Start with initials and see for each one if it's in `VoteCount`.